moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,176 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import argparse
+import os
+import moduleframework
+import tempfile
+import textwrap
+import sys
+
+from avocado.utils import process
+from moduleframework import common
+
+
+def cli():
+    file_name = os.path.basename(sys.argv[0])
+    parser = argparse.ArgumentParser(
+        # TODO
+        description=textwrap.dedent('''\
+        unknown arguments are forwarded to avocado:
+           optionally use additional avocado param like --show-job-log, see avocado action --help'''),
+        formatter_class=argparse.RawTextHelpFormatter,
+        # epilog(with http link) is used in some error msg too:
+        epilog="see http://meta-test-family.readthedocs.io for more info",
+        usage=("{0} [options] local_tests".format(file_name)),
+    )
+    parser.add_argument("--linter", "-l", action="store_true", default=False, help='adds additional compose checks')
+    parser.add_argument("--setup", action="store_true", default=False, help='Setup by mtfenvset')
+    parser.add_argument("--action", action="store", default='run', help='Action for avocado, see avocado --help for subcommands')
+
+    # some useful bash variables
+    # there are more possible variables up to the doc, not sure what could be like options too
+    group = parser.add_argument_group('additional arguments are like environment variables up to the http://meta-test-family.readthedocs.io/en/latest/user_guide/environment_variables.html ')
+    group.add_argument("--module", action="store", help='Module type, like: docker, nspawn or rpm')
+    group.add_argument("--debug", action="store_true", help='passes AVOCADO_LOG_DEBUG=yes DEBUG=yes')
+    group.add_argument("--config", action="store", help='CONFIG=config.yaml')
+    group.add_argument("--url", action="store", help='URL overrides the value of module.docker.container or module.rpm.repo.')
+    group.add_argument("--modulemdurl", action="store", help='overwrites the location of a moduleMD file')
+
+    # unknown options are forwarded to avocado run
+    args, unknown = parser.parse_known_args()
+
+    common.print_debug("Options: linter={0}, setup={1}, action={2}, module={3}".format(
+        args.linter, args.setup, args.action, args.module))
+    common.print_debug("remaining options for avocado: {0}".format(unknown))
+
+    # environment usage:
+    #   read: os.environ.get('MODULE')
+    #   write: os.environ['MODULE'] 
+
+    # MODULE could be from:
+    #   1. common.get_module_type() ... it reads config.yaml and treceback if it doesn't exist
+    #   2. environment ... MODULE=docker etc
+    #   3. argument ... --module=docker
+    try:
+        args.module=common.get_module_type()
```

Comment:
please check your code and repair all troubles like this
there has to be `A = B`

Reply:
done, autopep8 -i